src/main/java/li/cil/sedna/utils/HashSetUtils.java

```diff
@@ -0,0 +1,11 @@
+package li.cil.sedna.utils;
+
+import java.util.HashSet;
+
+public final class HashSetUtils {
+    public static <E> HashSet<E> hashSetForSize(int size) {
+        final float loadFactor = (2.0f/3);
```

Comment:
What's the reasoning for this helper, instead of, say, just `new HashSet<E>(size)`? Did you do some tests and it performed better?

Reply:
Ah yes. So the Java API for HashSets is dreadful. You would think the "initialCapacity" for a HashSet would be the number of elements you can put in before it has to resize. But that's not what it means. If you pass a `size` and then add `size` elements, the data structure will have to immediately resize, reallocing and rehashing everything, part way through adding them all. This is because a HashSet resizes once the internal capacity reaches a percentage of loadFactor or higher. A really bad decision on Sun's part to expose an implementation detail like that.
So this just creates a HashSet of the right size to hold all the elements.

Reply:
Ahh, I see. Might be reasonable to change this to `it.unimi.dsi.fastutil.longs.LongArraySet`? Doesn't have that quirk (from what I can tell) and would avoid boxing (when switching to `LongSet` instead of `Set<Long>` in the TLB and interpret args)?

Reply:
That's perfect! The boxing overhead always bothered me. Pushed a commit